flink-runtime/src/test/java/org/apache/flink/runtime/rpc/RpcSSLAuthITCase.java

```diff
@@ -79,55 +77,62 @@ public void testConnectFailure() throws Exception {
         sslConfig2.setString(SecurityOptions.SSL_INTERNAL_TRUSTSTORE_PASSWORD, "password");
         sslConfig2.setString(SecurityOptions.SSL_ALGORITHMS, "TLS_RSA_WITH_AES_128_CBC_SHA");
 
-        RpcService rpcService1 = null;
-        RpcService rpcService2 = null;
-
-        try {
-            // to test whether the test is still good:
-            //   - create actorSystem2 with sslConfig1 (same as actorSystem1) and see that both can
-            // connect
-            //   - set 'require-mutual-authentication = off' in the ConfigUtils ssl config section
-            rpcService1 =
-                    RpcSystem.load()
-                            .localServiceBuilder(sslConfig1)
-                            .withBindAddress("localhost")
-                            .withBindPort(0)
-                            .createAndStart();
-            rpcService2 =
-                    RpcSystem.load()
-                            .localServiceBuilder(sslConfig2)
-                            .withBindAddress("localhost")
-                            .withBindPort(0)
-                            .createAndStart();
-
-            TestEndpoint endpoint = new TestEndpoint(rpcService1);
-            endpoint.start();
-
-            CompletableFuture<TestGateway> future =
-                    rpcService2.connect(endpoint.getAddress(), TestGateway.class);
-            TestGateway gateway = future.get(10000000, TimeUnit.SECONDS);
-
-            CompletableFuture<String> fooFuture = gateway.foo();
-            fooFuture.get();
-
-            fail("should never complete normally");
-        } catch (ExecutionException e) {
-            // that is what we want
-            assertTrue(e.getCause() instanceof RpcConnectionException);
-        } finally {
-            final CompletableFuture<Void> rpcTerminationFuture1 =
-                    rpcService1 != null
-                            ? rpcService1.closeAsync()
-                            : CompletableFuture.completedFuture(null);
-
-            final CompletableFuture<Void> rpcTerminationFuture2 =
-                    rpcService2 != null
-                            ? rpcService2.closeAsync()
-                            : CompletableFuture.completedFuture(null);
-
-            FutureUtils.waitForAll(Arrays.asList(rpcTerminationFuture1, rpcTerminationFuture2))
-                    .get();
-        }
+        assertThatThrownBy(
```

Comment:
IMO it makes the test more readable to only include the code that throws the exception inside the `assertThatThrownBy` call:
```java
            assertThatThrownBy(
                            () -> {
                                TestGateway gateway = future.get(10000000, TimeUnit.SECONDS);

                                CompletableFuture<String> fooFuture = gateway.foo();
                                fooFuture.get();
                            })
                    .withFailMessage("should never complete normally")
                    .isInstanceOf(ExecutionException.class)
                    .hasCauseInstanceOf(RpcConnectionException.class);
```